### PR TITLE
publish-commit-bottles: approve PRs after bottle push

### PR DESCRIPTION
**Pull request by carlocab**
We want to enable the dismissal of reviews after pushes to a PR branch,
but that means that approving reviews are dismissed after @BrewTestBot
pushes the bottle commit(s) to a PR.

To avoid PRs needing a second review before merging, let's have
@BrewTestBot approve the PR after the bottle push.

To support this, I've added a few checks for consistency of the HEAD
SHA1 in the runner's checkout and on the PR branch to avoid merging
commits we don't intend to. These checks are not airtight, but they're
better than nothing. We also check first that the PR is approved before
proceeding.

Finally: I've CCed myself in all comments posted after failures so I can
get a better look at making this workflow handle those situations
better.

- Revert "publish-commit-bottles: disable automerge"
- publish-commit-bottles: approve PRs after bottle push

### CODE REVIEW

**Comment by carlocab** [comment]
Tested successfully at #128384.

We probably want to enable dismissal of reviews on push before merging this.

**Review by ZhongRuoyu — APPROVED** [review]

**Comment by dawidd6** [comment]
Can BrewTestBot approve it's own PRs created by the autobump workflow?

**Comment by carlocab** [comment]
> Can BrewTestBot approve it's own PRs created by the autobump workflow?

No, it can't 😅

**Comment by carlocab** [comment]
Maybe we can switch the autobump workflow to use `GITHUB_TOKEN` so that the PR is created by the @github-actions bot?

**Comment by ZhongRuoyu** [comment]
> Maybe we can switch the autobump workflow to use `GITHUB_TOKEN` so that the PR is created by the @github-actions bot?

Is it possible to have @github-actions author a PR, though?

**Comment by carlocab** [comment]
> Is it possible to have @github-actions author a PR, though?
> 
> Update: I asked this because I haven't seen that before, but yes: #127137

It's possible, but I've forgotten why that's not a good idea: #127164

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
> We probably want to enable dismissal of reviews on push before merging this.

Enabled!

**Comment by carlocab** [comment]
> > We probably want to enable dismissal of reviews on push before merging this.
> 
> 
> 
> Enabled!

Thanks Mike!

**Comment by carlocab** [comment]
> Can BrewTestBot approve it's own PRs created by the autobump workflow?

Handled in #128697. We can have @github-actions approve the PR after bottle push.

**Pull request merged**
